scripts/toaster-layers-recipes-config.py

```diff
@@ -25,27 +23,129 @@ def get_layer_name(lpath):
     lc.close()
     return layer
 
-# Parse through layers from bblayers.conf and pick up layer names
-for base_layer in bblayers.split():
-    layer = get_layer_name(base_layer)
-    if layer:
-        if layer == OE_CORE_NAME:
-            continue
-        configured_layers.append(layer)
+bblayers = []
+bblayerconf = builddir + "/conf/bblayers.conf"
+start_found = False
+blc = open(bblayerconf, "r")
+for ln in blc:
+    if start_found:
+        if ln.startswith("\""):
+            break
+        bblayers.append(ln.strip().split()[0])
+    elif ln.startswith("BBLAYERS = \""):
+        start_found = True
+blc.close()
 
-totalLayers = list(configured_layers)
+localconf = builddir + "/conf/local.conf"
+lc = open(localconf, "r")
+distro_found, machine_found = False, False
+for ln in lc:
+    dist = re.match("DISTRO *\= *\'", ln)
+    if dist:
+        distro = ln.split("\'")[1]
+        distro_found = True
+    mch = re.match("MACHINE *\?\?\= *\"", ln)
+    if mch:
+        machine = ln.split("\"")[1]
+        machine_found = True
+    if distro_found and machine_found:
+        break
+lc.close()
 
-layer_and_recipes = OrderedDict()
-
-recipeListFile = builddir + "/conf/recipes-list.txt"
 customXML = builddir + "/custom.xml"
+fd = open(customXML, 'a')
```

Comment:
@sharidas, very valid and would make it more clear where the file is actually created at first. I'll make the change.